executor/executor.go

```diff
@@ -104,40 +104,19 @@ func copyLinesUntil(r io.Reader, w io.Writer, match string) (int, error) {
 	return ExitOk, nil
 }
 
-func doRunSetupCommand(emitter screwdriver.Emitter, f *os.File, r io.Reader, setupCommands []string) error {
+func doRunSetupCommand(guid string, emitter screwdriver.Emitter, f *os.File, r io.Reader, setupCommands []string) error {
 	var (
-		t      string
 		err    error
-		reader = bufio.NewReader(r)
-		reEcho = regexp.MustCompile("^$")
 	)
 
 	shargs := strings.Join(setupCommands, " && ")
 
 	f.Write([]byte(shargs))
 
-	t, err = readln(reader)
-	for err == nil {
-		t = strings.TrimLeft(t, "# ")
-		t = strings.TrimLeft(t, "$ ")
-		echoCmd := reEcho.FindStringSubmatch(t)
-		if len(echoCmd) != 0 {
-			_, werr := fmt.Fprintln(emitter, t)
-			if werr != nil {
-				return fmt.Errorf("Error piping logs to emitter: %v", werr)
-			}
-			return nil
-		}
-		_, werr := fmt.Fprintln(emitter, t)
-		if werr != nil {
-			return fmt.Errorf("Error piping logs to emitter: %v", werr)
-		}
-		t, err = readln(reader)
-	}
-	if err != nil {
-		return fmt.Errorf("Error with reader: %v", err)
-	}
-	return nil
+	// ignore exit code in setup commands
+	_, err = copyLinesUntil(r, emitter, guid)
```

Comment:
use copyLinesUntil same as [doRunCommand](https://github.com/screwdriver-cd/launcher/blob/ea288c72d7d367dcc8ab0c515f1d4ff94335e937/executor/executor.go#L135) to detect the end of commands.